apps/web/src/hooks/usePoolsOnChain.ts

```diff
@@ -68,7 +70,8 @@ function candidatePoolsOnChainHookFactory<TPool extends Pool>(
         SmartRouter.logger.metric(label)
         const getViemClients = createViemPublicClientGetter({ transportSignal: signal })
         const resolvedPairs = await SmartRouter.getPairCombinations(currencyA, currencyB)
-        const pools = await getPoolsOnChain(resolvedPairs ?? [], getViemClients, blockNumber)
+        const poolConfigs: StableSwapPool[] = await getStableSwapPools(currencyA?.chainId ?? ChainId.BSC)
+        const pools = await getPoolsOnChain(resolvedPairs ?? [], getViemClients, blockNumber, poolConfigs)
```

Comment:
can we fetch pool configs on the fly? Don't think it's necessary to update the interface here

Reply:
done